src/arena.c

```diff
@@ -744,11 +743,12 @@ arena_decay_init(arena_decay_t *decay, extents_t *extents, ssize_t decay_time,
 }
 
 static bool
-arena_decay_time_valid(ssize_t decay_time) {
-	if (decay_time < -1) {
+arena_decay_ms_valid(ssize_t decay_ms) {
+	if (decay_ms < -1) {
 		return false;
 	}
-	if (decay_time == -1 || (uint64_t)decay_time <= NSTIME_SEC_MAX) {
+	if (decay_ms == -1 || (uint64_t)decay_ms <= NSTIME_SEC_MAX *
+	    KQU(1000)) {
```

Comment:
That's how I originally wrote it, but this is more precise (no rounding error), and it does not risk overflow because ```NSTIME_SEC_MAX``` is guaranteed to be at least 1 billion times smaller than the full range of ```uint64_t```.